src/Button.tsx

```diff
@@ -0,0 +1,291 @@
+/** @jsx jsx */
+import { jsx } from "@emotion/core";
+import React from "react";
+import * as colors from "./colors";
+import * as CSS from "csstype";
+import { base } from "./typography";
+
+// Types that could use some improvement:
+// * Don't allow `children` and `icon` to be missing
+// * Don't allow `children` when `FAB`
+//
+// I was able to get guarantees to work, but only with very cryptic errors. I
+// decided it'd be best, for the time being, to `throw` if we use things
+// incorrectly.
+interface Props {
+  /**
+   * Background color to show in a non-disabled, non-active state
+   */
+  backgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Background color to show when button is being hovered over and is not disabled
+   */
+  hoverBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * The content to show inside the button
+   *
+   * When using an icon, use the `icon` prop instead.
+   */
+  children?: React.ReactNode;
+
+  /**
+   * Class name to passthrough to the top level component
+   */
+  className?: string;
+
+  /**
+   * Disable the button
+   *
+   * This will prevent all clicks and will show the button in a different color
+   */
+  disabled?: boolean;
+
+  /**
+   * Background color to display when the button is disabled
+   */
+  disabledBackgroundColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Text color to display when the button is disabled
+   */
+  disabledTextColor?: CSS.BackgroundColorProperty;
+
+  /**
+   * Treat this button as a floating action button
+   *
+   * You must include an `icon` prop and you must _not_ include a `children`
+   * prop for a floating action button.
+   */
+  fab?: boolean;
+
+  /**
+   * Force the button to be in the active state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceActiveState?: boolean;
+
+  /**
+   * Force the button to be in the focus state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceFocusState?: boolean;
+
+  /**
+   * Force the button to be in the hover state.
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceHoverState?: boolean;
+
+  /**
+   * A hidden button will have no background
+   *
+   * You should configure the `hiddenActiveBackgroundColor` prop if you are
+   * using any custom colors.
+   */
+  hidden?: boolean;
+
+  /**
+   * Color to make the background for a hidden button that is active.
+   *
+   * This color should be darker than the `hoverBackgroundColor`.
+   *
+   * This should not be used for a non-hidden button.
+   */
+  hiddenActiveBackgroundColor?: CSS.BackgroundColorProperty;
```

Comment:
Along with the previous comment, I guess I'm not sure why we need a separate set of colors just for the "hidden" style?  Like can't we just provide an `activeBackgroundColor` and `focusedTextColor`?  Do we also need a `textColor` property?

Reply:
Really good question.

The "active" state only changes for hidden buttons, not for regular buttons.

Note that the rest state's background color matches the active state's background color:

![image](https://user-images.githubusercontent.com/5224118/60850378-e1b53080-a1a3-11e9-928e-dd3b304d26b0.png)

Note here that the rest state's background color differs from the active state's background color:

![image](https://user-images.githubusercontent.com/5224118/60850390-f98cb480-a1a3-11e9-8233-92e105c5d73e.png)

Reply:
I guess I still don't understand why we would refuse to support an always-available  `activeBackgroundColor` and `focusedTextColor` property, feels like it could come in handy?

Reply:
You're totally right @timbotnik. We changed direction to have sensible defaults for the colors, but all other configuration for colors will have to be with either tailwind classes or emotion styles instead of props. That will reduce the surface area of the API dramatically.